u8timeseries/models/tcn_model.py

```diff
@@ -0,0 +1,254 @@
+            self.res_blocks_list.append(res_block)
+        self.res_blocks = nn.ModuleList(self.res_blocks_list)
+
+    def forward(self, x):
+        # data is of size (batch_size, input_length, input_size)
+        batch_size = x.size(0)
+        x = x.transpose(1, 2)
+
+        for res_block in self.res_blocks_list:
+            x = res_block(x)
+
+        x = x.transpose(1, 2)
+        x = x.view(batch_size, self.input_length, 1)
+
+        return x
+
+
+class TCNModel(TorchForecastingModel):
+
+    def __init__(self,
+                 input_length: int = 12,
+                 output_length: int = 1,
+                 kernel_size: int = 3,
+                 num_filters: int = 3,
+                 num_layers: Optional[int] = None,
+                 dilation_base: int = 2,
+                 weight_norm: bool = False,
+                 dropout: float = 0.2,
+                 **kwargs):
+
+        """ Temporal Convolutional Network Model (TCN).
+
+        This is an implementation of a dilated TCN used for forecasting.
+        Inspiration: https://arxiv.org/abs/1803.01271
+
+        Parameters
+        ----------
+        input_length
+            Number of past time steps that are fed to the forecasting module.
+        output_length
+            Number of time steps the torch module will predict into the future at once.
+        kernel_size
+            The size of every kernel in a convolutional layer.
+        num_filters
+            The number of filters in a convolutional layer of the TCN.
+        weight_norm
+            Boolean value indicating whether to use weight normalization.
+        dilation_base
+            The base of the exponent that will determine the dilation on every level.
+        num_layers
+            The number of convolutional layers.
+        dropout
+            The dropout rate for every convolutional layer.
+        """
+
+        raise_if_not(kernel_size < input_length,
+                     "The kernel size must be strictly smaller than the input length.", logger)
+        raise_if_not(output_length < input_length,
+                     "The output length must be strictly smaller than the input length", logger)
+
+        self.input_size = 1
+        kwargs['input_length'] = input_length
+
+        self.model = _TCNModule(input_size=self.input_size, input_length=input_length,
+                                kernel_size=kernel_size, num_filters=num_filters,
+                                num_layers=num_layers, dilation_base=dilation_base,
+                                output_length=output_length, dropout=dropout, weight_norm=weight_norm)
+
+        super().__init__(**kwargs)
+
+    def create_dataset(self, series):
+        return _TimeSeriesDataset1DShifted(series, self.input_length, self.output_length)
+
+    def predict(self, n: int) -> TimeSeries:
+        super().predict(n)
+
+        scaled_series = self.training_series.values()[-self.seq_len:]
```

Comment:
The name `scaled_series` was there in RNN for historical reasons (as the model was previously also scaling the series). We should use a better name like `input_sequence` or something

Reply:
done